addons/hr_homeworking/models/res_partner.py

```diff
@@ -5,21 +5,29 @@
 class ResPartner(models.Model):
     _inherit = 'res.partner'
 
+    def _get_main_employee(self):
+        self.ensure_one()
+        if self.is_company:
+            return self.env['hr.employee']
+        else:
+            return next((employee for employee in self.employee_ids if employee.company_id.id == self.env.company.id), self.env['hr.employee'])
     def _compute_im_status(self):
         super()._compute_im_status()
-        users = self.env["res.users"].search([
-            ('partner_id.id', 'in', self.ids)])
-        for user in users:
-            dayfield = self.env['hr.employee']._get_current_day_location_field()
-            location_type = user[dayfield].location_type
+        dayfield = self.env['hr.employee']._get_current_day_location_field()
+        for partner in self:
+            employee = partner._get_main_employee()
+            location_type = employee[dayfield].location_type
             if not location_type:
                 continue
-            im_status = user.partner_id.im_status
+            im_status = partner.im_status
             if im_status == "online" or im_status == "away" or im_status == "offline":
-                user.partner_id.im_status = location_type + "_" + im_status
+                partner.im_status = location_type + "_" + im_status
 
     def get_worklocation(self, start_date, end_date):
-        employee_id = self.env['hr.employee'].search([
-            ('work_contact_id.id', 'in', self.ids),
-            ('company_id.id', '=', self.env.company.id)])
-        return employee_id._get_worklocation(start_date, end_date)
+        employees = None
+        for partner in self:
+            if employees is None:
+                employees = partner._get_main_employee()
+            else:
+                employees += partner._get_main_employee()
```

Comment:
```suggestion
        employees = partner.employee_ids.filtered(lambda e: e.company_id == self.env.company)
```

Reply:
But a partner can be linked to multiple employees in the same company, so it's not possible to do that.